catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/RefreshRegistryEntries.java

```diff
@@ -189,27 +230,44 @@ private RemoteRegistryResults getRemoteRegistryMetacardsMap() throws FederationA
         } catch (Exception e) {
             throw new FederationAdminException("Error querying for local metacards ", e);
         }
-        SourceResponse response;
+        List<Callable<RemoteResult>> tasks = new ArrayList<>();
```

Comment:
Could we extract the 2 try/catch blocks and the for loop to separate private methods, or maybe encapsulate some of this logic into its own separate class? This method is fairly long and not all that easy to understand at first glance.